.drone/lcov-report.sh

```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
+
+set -e
+# set -x
+
+SKIPLIST="math"
+
+: "${LCOV_SKIP_PATTERN:='^[9]'}" # Set default lcov skip pattern
+
+pwd
+
+export CODECOV_SCRIPT=${BOOST_CI_SRC_FOLDER}/ci/travis/codecov.sh
+export CI_DIR=${BOOST_CI_SRC_FOLDER}/ci
+export BOOST_CI_CODECOV_IO_UPLOAD="skip"
+export LCOV_VERSION="v2.3"
+export LCOV_IGNORE_ERRORS_LEVEL=standard
+export EXPORT_BOOST_SRC_DIR="yes"
+
+touch /tmp/failed.txt
+touch /tmp/succeeded.txt
+
+cd "$BOOST_ROOT"
+git submodule update --init --recursive
+./b2 headers
+
+# The script runcodecov.sh will be pieced together in parts, enabling variables
+# to be included into the contents of the script.
+
+# shellcheck disable=SC2016
+textpart1='#!/bin/bash
+set -x
+reponame=$1
+echo "reponame is $reponame"
+echo "date is $(date)"
+mkdir -p /tmp/lcov-repo-results || true
+skiplist="'
+
+textpart2="${SKIPLIST}"
```

Comment:
To make this easier you could use either `text+=` or just change quotes like 
```
text='use=41"'\
"$variable"'
$non-evaled'
```

Or use a here-doc and sed with placeholders

Reply:
I was thinking about how to improve this string creation to be "clean".   
In the end I think it would be cleanest to NOT have this string at all:

- add `ci/runcodecov.sh` as a regular script file
- before `git submodule foreach` export the 2 required variables from this script (`SKIPLIST` & `LCOV_SKIP_PATTERN`)
- alternatively pass those as arguments for better visibility and fetch them at the start of the script

Having the script separate allows for easier development, linting (I see you also use shellcheck) and manual execution for focused testing

BTW: I'd suggest to always use `set -eu` at the top of the scripts to detect misspelled variables, missing parameters and unexpectedly failing commands